lib/cc/workspace/exclusion.rb

```diff
@@ -0,0 +1,29 @@
+module CC
+  class Workspace
+    class Exclusion
+      def initialize(pattern)
+        @pattern = simplify(pattern)
+      end
+
+      def expand
+        if glob?
+          Dir.glob(pattern)
+        else
+          [pattern]
+        end
+      end
+
+      def glob?
+        pattern.include?("*")
+      end
+
+      private
+
+      attr_reader :pattern
+
+      def simplify(pattern)
+        pattern.to_s.sub(%r{(/\*\*)?(/\*)?$}, "")
```

Comment:
I believe, because of this, we no longer need that `is_a?(Glob) => .value` hack. Since this will simplify `x/**/*` and `x/**` both to `x`. After adding the workspace spec using `vendor/**`, it still passes without that logic.